### PR TITLE
fix: [CDS-41389]: added runtime support in env V2 with BE support for azure web app

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

With the new env entity in place we don't have access to the fixed values that were set in the studio so we needed new API's to fetch those on the basis of env id and infra id

#### Screenshots

https://user-images.githubusercontent.com/106532291/184340347-7a13af58-5f51-4acd-a93d-687982b3ebdf.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 444 Issues 
* Code issue 444 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10578)

**Comment by vikrant-harness** [comment]
retrigger titlecheck

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/AzureWebAppInfrastructureStep/AzureWebAppInfrastructureSpecInputForm.tsx [hunk 1]
```diff
@@ -128,20 +144,57 @@ export const AzureWebAppInfrastructureSpecInputForm: React.FC<
...
+    loading: loadingResourceGroupsV2,
+    error: resourceGroupsErrorV2
+  } = useGetAzureResourceGroupsV2({
+    queryParams,
+    lazy: true
+  })
+
+  const fetchResourceUsingEnvId = (): boolean => {
+    return (
+      getMultiTypeFromValue(connector) !== MultiTypeInputType.RUNTIME &&
```
Comment (depanshu-dhiman) [inline src/modules/75-cd/components/PipelineSteps/AzureWebAppInfrastructureStep/AzureWebAppInfrastructureSpecInputForm.tsx:159]: I can see that `getMultiTypeFromValue(connector)` is getting called from multiple places, can this be stored in a variable and reused? Same for any other parameter like `subscriptionId`, `environmentRef`, etc

File: src/modules/75-cd/components/PipelineSteps/AzureWebAppInfrastructureStep/AzureWebAppInfrastructureSpecInputForm.tsx [hunk 2]
```diff
@@ -318,6 +371,19 @@ export const AzureWebAppInfrastructureSpecInputForm: React.FC<
                     }
                   })
                 }
+                if (fetchResourceUsingEnvId()) {
```
Comment (depanshu-dhiman) [inline src/modules/75-cd/components/PipelineSteps/AzureWebAppInfrastructureStep/AzureWebAppInfrastructureSpecInputForm.tsx:374]: Shouldn't this be an else if? Unless there is a case where `refetchResourceGroupsV2` and `refetchResourceGroups` both calls are required.
I don't have full context so idk if 2 calls are required or not, but this looks like there can be a condition when both APIs will be called.

**Review by depanshu-dhiman — CHANGES_REQUESTED** [review]
Minor comments

File: src/modules/75-cd/components/PipelineSteps/AzureWebAppInfrastructureStep/AzureWebAppInfrastructureSpecInputForm.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/AzureWebAppInfrastructureStep/AzureWebAppInfrastructureSpecInputForm.tsx:159]: @depanshu-dhiman changed this in the latest commit

File: src/modules/75-cd/components/PipelineSteps/AzureWebAppInfrastructureStep/AzureWebAppInfrastructureSpecInputForm.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/AzureWebAppInfrastructureStep/AzureWebAppInfrastructureSpecInputForm.tsx]: @depanshu-dhiman done

**Comment by vikrant-harness** [comment]
retrigger jest

**Comment by vikrant-harness** [comment]
retrigger jest

**Review by Eric761 — DISMISSED** [review]

**Comment by vikrant-harness** [comment]
trigger cypress

**Review by YounixIO — APPROVED** [review]

**Review by Sagar-Rajput7 — APPROVED** [review]

**Review by depanshu-dhiman — APPROVED** [review]

**Review by Sagar-Rajput7 — APPROVED** [review]

**Review by YounixIO — APPROVED** [review]

**Pull request merged**